src/test/java/com/google/api/codegen/testsrc/common/library.proto

```diff
@@ -921,14 +902,12 @@ message TestOptionalRequiredFlatteningParamsRequest {
   bytes required_singular_bytes = 8 [(google.api.field_behavior) = REQUIRED];
   InnerMessage required_singular_message = 9 [(google.api.field_behavior) = REQUIRED];
   string required_singular_resource_name = 10 [(google.api.field_behavior) = REQUIRED,
-    (google.api.resource_reference) = "Book"
+    (google.api.resource_reference).type = "library.googleapis.com/Book"
   ];
   string required_singular_resource_name_oneof = 11 [(google.api.field_behavior) = REQUIRED,
-    (google.api.resource_reference) = "google.example.library.v1.Book"
-  ];
-  string required_singular_resource_name_common = 14 [(google.api.field_behavior) = REQUIRED,
```

Comment:
What's the status of common resource names like `Project` and `Organization`?

Reply:
References to these types need to be through the ResourceDescriptor.child_type field. I don't know how to translate the xamples in library.proto into such references, since there isn't a pattern with format `projects/{project}/foos/{foo}`

Reply:
With this new spec, there is no way to just define a `projects/{project}`? This is necessary for many APIs (30ish)

Reply:
The `projects/{project}` name is implied if you specify a `.child_type` reference to a `projects/{project}/foos/{foo}` pattern.